### ISSUE TITLE
`transformer_asr.py`: incorrect `source_maxlen`

### ISSUE THREAD

**Issue by MicahDoo** [keras-team/keras-io#1389]
https://github.com/keras-team/keras-io/blob/master/examples/audio/transformer_asr.py

In the code at the above link, I found that `source_maxlen` is defaulted to `100` in the transformer.
The problem, though, is that the inputs are actually padded to length 2754, where it's then downsampled with CNN by a factor of 8. The result is a sequence of length 345, which is far greater than 2754.
Correct me if I am wrong, but I reckon that is a bug?

### Problem code:
In the transformer definition, `source_maxlen` is defaulted to 100:
```python
class Transformer(keras.Model):
    def __init__(
        self,
        num_hid=64,
        num_head=2,
        num_feed_forward=128,
        source_maxlen=100,
        target_maxlen=100,
        num_layers_enc=4,
        num_layers_dec=1,
        num_classes=10,
    ):
```
... which isn't explicitly set at instantiation:
```python
model = Transformer(
    num_hid=200,
    num_head=2,
    num_feed_forward=400,
    target_maxlen=max_target_len,
    num_layers_enc=4,
    num_layers_dec=1,
    num_classes=34,
)
```

**Comment by sachinprasadhs** [comment]
@apoorvnandan, would you be able to help in the above issue related to your published tutorial here https://keras.io/examples/audio/transformer_asr/

**Comment by apoorvnandan** [comment]
Hi! Just saw this.

On a cursory glance, it does look like a bug. 

1. The param name is misleading. That param is used to determine the `input_dim` in the `Embedding` layer. which should be something liek 129. (which comes from the stft of audio)
2. I think it should be explicity set to the above value instead of letting it default.

I'm not a 100% sure though. Will try to go through this after work to check if there is something I missed.

**Comment by maitry63** [comment]
Hi @MicahDoo, thank you for your question.

@apoorvnandan, after reviewing the implementation in detail, I believe there may be a small misunderstanding regarding `source_maxlen`.

In the current encoder design, positional embeddings or a fixed-length embedding table are not used. Instead, the encoder relies on Conv1D layers with stride for downsampling, meaning the sequence length is dynamically determined by the convolution output.

As a result, `source_maxlen` is not referenced anywhere in the encoder path and does not impact model computation.

It seems this parameter may be a remnant from an earlier version of the example, where positional embeddings might have been included.

Additionally, the STFT feature dimension (~129) corresponds to the frequency axis of the spectrogram. This is independent of sequence length and therefore unrelated to source_maxlen.

Given this, the safest approach would likely be to remove `source_maxlen` altogether, unless positional embeddings are reintroduced in the future.

For reference, please see this notebook:
https://colab.sandbox.google.com/gist/maitry63/5396fb648affdfb50373475f61bf115a/transformer_asr.ipynb#scrollTo=Kwo5OHOtFz5D

I will also raise a PR for this change.

Please let me know if this helps!
Thank you.
